poke_redirect.py

```diff
@@ -0,0 +1,277 @@
+#!/usr/bin/env python3
+
+import argparse
+import collections
+import operator
+import sys
+
+from typing import Dict
+
+# Various constants for dealing with IPS files
+PATCH_BYTEORDER = 'big'
+PATCH_HEADER = b'PATCH'
+PATCH_EOF = b'EOF'
+
+# Both pokear and salmon use the same code offset; from the cxis with ctrtool
+CODE_OFFSET = 0x100000
+CODE_BYTEORDER = 'little'
+
+# Offset from the start of card_id function to where we skip from
+CARD_ID_SKIP_OFFSET = 0x34
+# Offset from the start of the card_id frunciton to where we fake the id
+CARD_ID_FAKE_OFFSET = 0x10c
+
+# Offsets from the end of the binary at which to install the SD functions
+# These values are based on the pre-built poke radar patch
+SD_FUNCTION_OFFSET = 0x310
+OPEN_FUNCTION_OFFSET = 0x3c8
+
+class Config(collections.namedtuple('AbstractConfig', (
+    'code_size', 'check_card', 'card_id', 'read_save', 'write_save',
+    'fs_handle', 'fs_open', 'fs_read', 'fs_write', 'fs_close', 'flip_write'))):
+  """Configuration for the Pokemon binary.
+
+  Attributes:
+    code_size: The size of the code .text section; obtained from cxi with
+      ctrtool.
+    check_card: The address of the function that checks if a card is inserted,
+      or none if the binary doesn't explicitly check for a card;
+      indirectly calls FSUSER_CardSlotIsInserted.
+    card_id: The address of the function that gets the card id;
+      indirectly calls FSUSER_GetLegacyRomHeader2.
+    read_save: The address of the function that reads the save data;
+      see TWLSaveTool SPIReadSaveData.
+    write_save: The address of the function that writes the save data;
+      see TWLSaveTool SPIWriteSaveData.
+    fs_handle: The address of the handle to the fs:USER process.
+    fs_open: The address of FSUSER_OpenFileDirectly.
+    fs_read: The address of FSFILE_Read.
+    fs_write: The address of FSFILE_Write.
+    fs_close: The address of FSFILE_Close.
+    flip_write: Indicates that the order of the 3rd and 4th arguments of the
+      write function are flipped.
+  """
+
+# Config for Dream Radar
+POKEAR_CONFIG = Config(
+  code_size  = 0x0bebb4,
+  # limit    = 0x0bf000
```

Comment:
Just to make sure I understand, is this for documentation purposes?

Reply:
Yep, that's just the size of all of the code pages, I just put that there for documentation

Reply:
Ah cool